### PR TITLE
Fix texture serialization bug.

### PR DESCRIPTION
**Pull request by roeas**

### CODE REVIEW

File: private/Scene/SceneDatabaseImpl.h [hunk 1]
```diff
@@ -125,11 +125,6 @@ class SceneDatabaseImpl
 			AddMesh(Mesh(inputArchive));
 		}
 
-		for (uint32_t meshIndex = 0; meshIndex < meshCount; ++meshIndex)
-		{
-			AddMesh(Mesh(inputArchive));
-		}
-
```
Comment (Hinageshi01) [inline private/Scene/SceneDatabaseImpl.h:132]: Are these redundant?

File: private/Scene/TextureImpl.h [hunk 2]
```diff
@@ -75,7 +75,7 @@ class TextureImpl final
 	template<bool SwapBytesOrder>
 	const TextureImpl& operator>>(TOutputArchive<SwapBytesOrder>& outputArchive) const
 	{
-		outputArchive << GetID().Data() << static_cast<uint8_t>(GetType()) << GetPath() << m_rawTexture.empty();
```
Comment (Hinageshi01) [inline private/Scene/TextureImpl.h:78]: Seems to be a bug.

File: private/Scene/TextureImpl.h [hunk 3]
```diff
@@ -75,7 +75,7 @@ class TextureImpl final
...
+		outputArchive << GetID().Data() << static_cast<uint8_t>(GetType()) << GetPath() << !m_rawTexture.empty();
```
Comment (roeas) [inline private/Scene/TextureImpl.h:78]: Seems to be a bug.

File: private/Scene/SceneDatabaseImpl.h (hunk 1 above)
Comment (roeas) [inline private/Scene/SceneDatabaseImpl.h:132]: Are these redundant?

Comment (T-rvw) [reply private/Scene/SceneDatabaseImpl.h:132]: Yes.👍 After git blame, I find that it is a mistake in my previous mesh bone data PR.

File: private/Scene/TextureImpl.h (hunk 3 above)
Comment (T-rvw) [reply private/Scene/TextureImpl.h:78]: Looks like a bug. @ChaoSXDemon

**Review by T-rvw — APPROVED** [review]

**Pull request merged**